openff/toolkit/utils/exceptions.py

```diff
@@ -1,48 +1,45 @@
-__all__ = (
-    "MessageException",
-    "IncompatibleUnitError",
-    "MissingDependencyError",
-    "MissingPackageError",
-    "ToolkitUnavailableException",
-    "LicenseError",
-    "InvalidToolkitError",
-    "InvalidToolkitRegistryError",
-    "UndefinedStereochemistryError",
-    "GAFFAtomTypeWarning",
-    "ChargeMethodUnavailableError",
-    "IncorrectNumConformersError",
-    "IncorrectNumConformersWarning",
-    "ChargeCalculationError",
-    "InvalidIUPACNameError",
-    "AntechamberNotFoundError",
-    "SMILESParseError",
-)
-
-# =============================================================================================
-# COMMON EXCEPTION TYPES
-# =============================================================================================
-
-
-class MessageException(Exception):
-    """A base class for exceptions that print out a string given in their constructor"""
+import warnings
+
+deprecated_names = ["MessageException", "ParseError"]
+
+
+def __getattr__(name):
+    if name in deprecated_names:
+        warnings.filterwarnings("default", category=DeprecationWarning)
+        warning_msg = f"{name} is DEPRECATED and will be removed in a future release of the OpenFF Toolkit."
+        if name == "MessageException":
+            warning_msg += (
+                " All custom exceptions now inherit from OpenFFToolkitException, "
+                "which should be used as a replacement for MessageException. Import it via "
+                "`from openff.toolkit.utils.exceptions import OpenFFToolkitException`."
+            )
+        warnings.warn(warning_msg, DeprecationWarning)
+        return globals()[f"_Deprecated{name}"]
+    raise AttributeError(f"module {__name__} has no attribute {name}")
+
+
+class _DeprecatedMessageException(Exception):
+    """DEPRECATED: A base class for exceptions that print out a string given in their constructor"""
```

Comment:
IMO this should have the same functionality it used to, given that it's still supported albeit in deprecated status.

```suggestion
    """DEPRECATED: A base class for exceptions that print out a string given in their constructor"""
    
   def __init__(self, msg):
        super().__init__(self, msg)
        self.msg = msg

    def __str__(self):
        return self.msg
```

Then `_DeprecatedParseError` can inherit from `_DeprecatedMessageException`, with the added bonus that I *think* that'll get rid of the LGTM warning about `__init__`.